algorithms/linfa-tsne/examples/tsne.rs

```diff
@@ -5,7 +5,7 @@ use std::{io::Write, process::Command};
 
 fn main() -> Result<()> {
     let ds = linfa_datasets::iris();
-    let ds = Pca::params(3).whiten(true).fit(&ds).transform(ds);
+    let ds = Pca::params(3).whiten(true).fit(&ds).unwrap().transform(ds);
```

Comment:
~~unwrap~~ -> ?

Reply:
While we are at it, I think it would be good to run the examples in the CI too, in the last PR I caught a similar error on a decision tree example I believe and these kinds of errors go unnoticed during testing. I can try to add that in this PR if it makes sense

Reply:
yeah you could pass `--examples` to the build workspace

Reply:
I would need to add an error variant for PCA error to the t-sne error struct in order to do this, do you think it is reasonable or should I leave `unwrap`?